src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -26,11 +35,93 @@ public static function onShowSearchHitTitle(
 		array &$query,
 		array &$attributes
 	): void {
-		if ( $title->getNamespace() !== WB_NS_ITEM ) {
+		$itemId = self::getItemId( $title );
+
+		if ( $itemId === null ) {
+			return;
+		}
+
+		$pageTitle = self::getItemPage( $itemId );
+
+		if ( $pageTitle === null ) {
 			return;
 		}
 
-		// TODO: get item site link and replace $title
+		$title = $pageTitle;
```

Comment:
For EntityResult results we just need to point to the new page, since the correct $titleSnippet gets set in `CirrusShowSearchHitHandler::onShowSearchHitTitle()`.